main.tf

```diff
@@ -74,7 +74,7 @@ data "tls_public_key" "public_key" {
 locals {
   # Network ip ranges
   vnet_cidr_block          = "192.168.0.0/16"
-  aks_subnet_cidr_block    = "192.168.1.0/24"
+  aks_subnet_cidr_block    = "192.168.16.0/20"
```

Comment:
This value should be adjusted based on the user's selection. If the network is `kubenet` then this should keep its old value. However, if it's `azure` then it should be assigned this value. Also what is driving this actual CIDR range: 192.168.16.0./20 why not 192.168.4.0/20 Just wondering why the large leap? This does not collide with any of the previous /24 items.

Reply:
@thpang just wondering why we would want to have a smaller subnet for "kubenet". We have the whole VPC at our disposal, and we are using only a tiny fraction of it. If anything, we should make the subnets larger to make use of the available ip range space so users don't run into issues if they happen to have a larger number of nodes, say.

Reply:
I can see your point. Currently the total hosts that can be on the AKS subnet is 256. I don't disagree there is always room for growth. With the new CIDR range the AKS subnet will have access to 4096 hosts. If we're going to adjust this value here we should also adjust the same values in the other providers. My suggestion would be to go with: 192.168.4.0/20 to keep things aligned. I am guessing we just took the 192.168.16.0/20 block from Barb's work.

Reply:
Using the CIDR range based on Azure error message. To @enderm comment it could be opened up to larger range.

Reply:
yeah, cidr range calculations are so not intuitive. I found this site helpful for avoiding range overlaps: http://www.davidc.net/sites/default/subnets/subnets.html

Reply:
Ok, then best to keep the 192.168.16.0/20 block. Great tool @enderm Very helpful. Also might need text on this in the code and make sure we propagate this into the other providers.